packages/core/src/utils/editor.ts

```diff
@@ -166,14 +259,61 @@ export async function openDiff(
         const command =
           process.platform === 'win32'
             ? `${diffCommand.command} ${diffCommand.args.join(' ')}`
-            : `${diffCommand.command} ${diffCommand.args.map((arg) => `"${arg}"`).join(' ')}`;
+            : `${diffCommand.command} ${diffCommand.args
+                .map((arg) => `"${arg}"`)
+                .join(' ')}`;
         execSync(command, {
           stdio: 'inherit',
           encoding: 'utf8',
         });
         break;
       }
 
+      case 'editor': {
+        // Handle VISUAL/EDITOR environment variable with proper execution strategy
+        const editorCmd = process.env.VISUAL || process.env.EDITOR;
+        if (!editorCmd) {
+          throw new Error('VISUAL or EDITOR environment variable not set');
+        }
+        const editorName = editorCmd.split(' ')[0].toLowerCase();
+
+        // Use spawn for GUI editors that might support --wait or similar
```

Comment:
Is there a better way to do this than special casing? If we're making this configurable IMO we should be relying entirely on configuration.